webcrypto/hmac.go

```diff
@@ -285,28 +285,28 @@ func (hip *HMACImportParams) ImportKey(
 	}
 
 	// 5. 6.
-	length := int64(len(keyData) * 8)
+	length := byteLength(len(keyData)).asBitLength()
 	if length == 0 {
 		return nil, NewError(DataError, "key length cannot be 0")
 	}
 
 	// 7.
 	if hip.Length.Valid {
-		if hip.Length.Int64 > length {
+		if hip.Length.Int64 > int64(length) {
 			return nil, NewError(DataError, "key length cannot be greater than the length of the imported data")
 		}
 
-		if hip.Length.Int64 < length {
+		if hip.Length.Int64 < int64(length) {
 			return nil, NewError(DataError, "key length cannot be less than the length of the imported data")
 		}
 
-		length = hip.Length.Int64
+		length = bitLength(hip.Length.Int64)
```

Comment:
Looking at this code:
1. do we need the last assignement? If it is not less and it is not more ... and it is an int* isn't it already equal?
2. Can we maybe make this just check that they are not equal - I don't think that it matters all that much to tell the user whether it is smaller or bigger.

Reply:
I strictly followed the steps of the [specification](https://www.w3.org/TR/WebCryptoAPI/#hmac-operations) there. But you're right in the end it doesn't make a big difference in that case. I've reorganized the code accordingly :+1:.